### PR TITLE
Incorporate ALS fixes (IVS-186)

### PR DESCRIPTION
**Pull request by civilx64**
1. Upgrade to newer version of ifcopenshell to avoid bug in ALS016 related to clothoid spirals in imperial units.

2. Adjust for the total length of vertical segments so that ALS017 checks are performed correctly.

(IVS-186, IVS-53)

### CODE REVIEW

**Review by Ghesselink — CHANGES_REQUESTED** [review]
- Can you check whether you checked out the right commit from development? The [updates made in the ALB012 branc](https://github.com/buildingSMART/ifc-gherkin-rules/pull/300/commits/afdc400d3aaf2c45e9378a06d7372e8839c9a91e)h are modified along with these commits. 
- I'm getting passed results for ALS017. Could this be that this has to do with the ifcopenshell version? I see the latest release of IfcOpenShell is 0.8.0, where do I find 0.8.1?

Edit; I've manually updated my conda environment to python3.12 and [ifcopenshell commit 92b63a0](https://github.com/IfcOpenBot/IfcOpenShell/commit/92b63a0), but still with the same result :(
image

**Comment by Ghesselink** [comment]
There are still some schema errors regarding the conversion. I'll also have a look 
image
(e.g. for pass-als017-scenario01-imperial_continuous.ifc)

File: features/environment.py [hunk 1]
```diff
@@ -63,19 +63,22 @@ def get_validation_outcome_hash(obj):
...
+
     old_outcomes = getattr(context, 'gherkin_outcomes', [])
     while context._stack[0].get('@layer') == 'attribute':
         context._pop()
     # preserve the outcomes to be serialized to DB in after_feature()
     context.gherkin_outcomes = old_outcomes
-    context.scenario_outcome_state[len(context.gherkin_outcomes)] = {'scenario': scenario.name,
-                                                     'last_step': scenario.steps[-1]}
-    
+    context.scenario_outcome_state.append(
```
[the remark this answers is not in the document]
Comment (civilx64) [reply features/environment.py:76]: The commit you noted, afdc400d3aaf2c45e9378a06d7372e8839c9a91e, was applied but my merge of ALB012 stomped over it.  I have cherry-picked it back to the top of the log:

image

**Comment by civilx64** [comment]
> I'm getting passed results for ALS017. Could this be that this has to do with the ifcopenshell version? I see the latest release of IfcOpenShell is 0.8.0, where do I find 0.8.1?

> Edit; I've manually updated my conda environment to python3.12 and [ifcopenshell commit 92b63a0](https://github.com/IfcOpenBot/IfcOpenShell/commit/92b63a0), but still with the same result :(

Per slack discussion, you need to update the `.venv` on your local copy of the platform which you can do with the makefiles in [validate/34be62b94106d96d00b723e3b1e2fef32effb6af](https://github.com/buildingSMART/validate/tree/34be62b94106d96d00b723e3b1e2fef32effb6af)

**Comment by Ghesselink** [comment]
> Per slack discussion, you need to update the .venv on your local copy of the platform which you can do with the makefiles in [validate/34be62b94106d96d00b723e3b1e2fef32effb6af](https://github.com/buildingSMART/validate/tree/34be62b94106d96d00b723e3b1e2fef32effb6af)

Yes, indeed. I thought I was not using venv anymore as I switched to have my own conda environment for running the server, but didn't realize venv was still running for the worker

**Pull request merged**